### ISSUE TITLE
How to measure the accuracy? F-score is there, why is that not included in th tf.summary?

### ISSUE THREAD

**Issue by QaisarRajput** [argman/EAST#92]
Hi, 
Great implementation! i have trained the model and now i want to measure the accuracy of the model. i see in `multigpu_train.py` and as you have talked about F-Score

        tf.summary.image('input', images)
        tf.summary.image('score_map', score_maps)
        tf.summary.image('score_map_pred', f_score * 255)   <-- f_score
        tf.summary.image('geo_map_0', geo_maps[:, :, :, 0:1])
        tf.summary.image('geo_map_0_pred', f_geometry[:, :, :, 0:1])
        tf.summary.image('training_masks', training_masks)
        tf.summary.scalar('model_loss', model_loss)
        tf.summary.scalar('total_loss', total_loss)

can this be used ? if yes, then how?

In addition to that. if we test our images and using output region files. calculate the overlap of ground truth bounding box and the predicted bounding box (true_positive) . and calculate false_positive, false_negative, true_negative too. Will that be an overkill or is it already somehow done in the code already?

**Comment by argman** [comment]
its inconvenient to compute f1 with tensorflow

**Comment by QaisarRajput** [comment]
Any other suggestions. Actually now that i have trained the model i want to evaluate on the testing data. I was looking into IoU (contrary to dice loss which is used while training). But the challenge in our case is that the output regions may be rotated. Hence cannot be explained with top-left and bottom-right two points and if we do, the precision of regions wouldn't be evaluated correctly. We don't have the angle of rotation either. What direction should be taken here. ultimately i am looking for some accuracy measure.

**Comment by zxytim** [comment]
You could use https://github.com/Toblerity/Shapely to compute IoU of two
polygons

**Comment by argman** [comment]
use opencv to create a mask is another option

**Comment by QaisarRajput** [comment]
i looked in to shapely. and was able to do it by the following code snippet (i will make a pull request later. Although, i think similar function is already written in the code some where). Thanks
```
def polygon_from_str(poly_str):
  """
  Create a shapely polygon object from gt or dt line.
  this function makes the shapely polygon object to be used in the later function. 
  input here is each line in "img_*.txt" files of the dataset
  """
  polygon_points = [float(o) for o in poly_str.split(',')[:8]]
  polygon_points = np.array(polygon_points).reshape(4, 2)
  polygon = Polygon(polygon_points).convex_hull
  return polygon

def polygon_iou(poly1, poly2):
  """
  Intersection over union between two shapely polygons.
  """
  if not poly1.intersects(poly2): # this test is fast and can accelerate calculation
    iou = 0
  else:
    try:
      inter_area = poly1.intersection(poly2).area
      union_area = poly1.area + poly2.area - inter_area
      iou = float(inter_area) / union_area
    except shapely.geos.TopologicalError:
      print('shapely.geos.TopologicalError occured, iou set to 0')
      iou = 0
  return iou
```

**Comment by cbasavaraj** [comment]
@QaisarRajput Hi, did you make a pull request with evaluations functions added? Thanks
